parser/src/main/java/com/facebook/coresql/sqllogictest/SqlLogicTest.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.sqllogictest;
+
+import com.facebook.coresql.parser.AstNode;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.text.DecimalFormat;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static com.facebook.coresql.parser.Unparser.unparse;
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+public final class SqlLogicTest
+{
+    private static AcceptedRdms currRdms;
+    private static Set<AcceptedRdms> skipIfSet;
+    private static AcceptedRdms onlyIf;
+    private static BufferedWriter writer;
+    private static FileVisitor<Path> fileVisitor;
+    private static Path startingDirectory;
+    private static int maxFilesToVisit;
+    private static int numFilesVisited;
+    private static int numStatementsParsed;
+    private static int numStatementsFound;
+
+    private enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ERROR
+    }
+
+    private enum AcceptedRdms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    private SqlLogicTest() {}
+
+    public static void run()
+            throws IOException
+    {
+        setup("all");
+        Files.walkFileTree(startingDirectory, fileVisitor);
+        shutdown();
+    }
+
+    public static void run(String rdms)
+            throws IOException
+    {
+        setup(rdms);
+        Files.walkFileTree(startingDirectory, fileVisitor);
+        shutdown();
+    }
+
+    private static void setup(String rdms)
+            throws IOException
+    {
+        verifyAndSetRdms(rdms);
+        createFileForResults();
+        writer = new BufferedWriter(new FileWriter("src/main/java/com/facebook/coresql/sqllogictest/sqllogictest_results.txt"));
+        startingDirectory = Paths.get("src/main/java/com/facebook/coresql/sqllogictest/test_statements");
```

Comment:
Are the test cases merged? Should it be included in the PR? They probably should live a `src/main/resources` instead? Also it would be nice to take this as a config parameter, or from commadline args.

Reply:
Adding the test cases seems to break the PR (I tried this before and the "changes" section wouldn't load because the test files have so many lines). So, I think we can leave it out and let the user point to the file containing the test cases on their machine.

I definitely agree with the config idea -- I've implemented a config.properties file in the latest push.

Done.